sklearn/compose/_column_transformer.py

```diff
@@ -494,8 +522,29 @@ def _validate_remainder(self, X):
         self._n_features = X.shape[1]
         cols = set(chain(*self._transformer_to_input_indices.values()))
         remaining = sorted(set(range(self._n_features)) - cols)
-        self._remainder = ("remainder", self.remainder, remaining)
         self._transformer_to_input_indices["remainder"] = remaining
+        remainder_cols = self._get_remainder_cols(remaining)
+        self._remainder = ("remainder", self.remainder, remainder_cols)
+
+    def _get_preferred_remainder_cols_dtype(self):
+        dtype = "int"
+        try:
+            all_dtypes = {_determine_key_type(c) for (*_, c) in self.transformers}
+            if len(all_dtypes) == 1:
+                dtype = list(all_dtypes)[0]
+        except ValueError:
+            pass
+        return dtype
```

Comment:
I assume we can remove this one and have an `else` inside the `if len(all_dtypes) == 1`